File: examples/pascal/05_loops.pas
{
  Loops - For, While, Repeat-Until
  
  Demonstrates:
  - For loop (To and DownTo)
  - While loop
  - Repeat-Until loop
  - Loop control
}

Program Loops;
Var
  i, sum, number: Integer;

Begin
  { FOR LOOP - counting 1 to 10 }
  WriteLn('=== For Loop: Count to 10 ===');
  For i := 1 To 10 Do
    Write(i, ' ');
  WriteLn;
  WriteLn;
  
  { FOR LOOP - countdown }
  WriteLn('=== Countdown ===');
  For i := 10 DownTo 1 Do
    Write(i, '... ');
  WriteLn('Liftoff!');
  WriteLn;
  
  { FOR LOOP - sum of numbers }
  WriteLn('=== Sum of 1 to 100 ===');
  sum := 0;
  For i := 1 To 100 Do
    sum := sum + i;
  WriteLn('Sum: ', sum);
  WriteLn;
  
  { WHILE LOOP - multiplication table }
  WriteLn('=== 7 Times Table ===');
  i := 1;
  While i <= 12 Do
  Begin
    WriteLn('7 × ', i, ' = ', 7 * i);
    i := i + 1;
  End;
  WriteLn;
  
  { REPEAT-UNTIL LOOP - input validation }
  WriteLn('=== Input Validation ===');
  Repeat
    Write('Enter a positive number: ');
    ReadLn(number);
    If number <= 0 Then
      WriteLn('That''s not positive! Try again.');
  Until number > 0;
  WriteLn('Thank you! You entered: ', number);
  WriteLn;
  
  { Nested loops - pattern }
  WriteLn('=== Star Pattern ===');
  For i := 1 To 5 Do
  Begin
    For number := 1 To i Do
      Write('* ');
    WriteLn;
  End;
End.

File: Examples/c/05_loops.c
/*
 * Loops - for, while, do-while
 * 
 * Demonstrates:
 * - for loop (counting)
 * - while loop (condition-based)
 * - do-while loop (execute at least once)
 * - break and continue
 */

#include <stdio.h>

int main() {
    int i, sum;
    
    // FOR LOOP - counting 1 to 10
    printf("=== For Loop: Count to 10 ===\n");
    for (i = 1; i <= 10; i++) {
        printf("%d ", i);
    }
    printf("\n\n");
    
    // FOR LOOP - countdown
    printf("=== Countdown ===\n");
    for (i = 10; i >= 1; i--) {
        printf("%d... ", i);
    }
    printf("Liftoff!\n\n");
    
    // FOR LOOP - sum of numbers
    printf("=== Sum of 1 to 100 ===\n");
    sum = 0;
    for (i = 1; i <= 100; i++) {
        sum += i;
    }
    printf("Sum: %d\n\n", sum);
    
    // WHILE LOOP - multiplication table
    printf("=== 7 Times Table ===\n");
    i = 1;
    while (i <= 12) {
        printf("7 × %d = %d\n", i, 7 * i);
        i++;
    }
    printf("\n");
    
    // DO-WHILE LOOP - input validation
    int number;
    printf("=== Input Validation ===\n");
    do {
        printf("Enter a positive number: ");
        scanf("%d", &number);
        if (number <= 0) {
            printf("That's not positive! Try again.\n");
        }
    } while (number <= 0);
    printf("Thank you! You entered: %d\n\n", number);
    
    // BREAK and CONTINUE
    printf("=== Even Numbers (using continue) ===\n");
    for (i = 1; i <= 20; i++) {
        if (i % 2 != 0) {
            continue;  // Skip odd numbers
        }
        printf("%d ", i);
    }
    printf("\n\n");
    
    printf("=== Find First Multiple of 7 (using break) ===\n");
    for (i = 1; i <= 100; i++) {
        if (i % 7 == 0) {
            printf("Found: %d\n", i);
            break;  // Stop after finding first one
        }
    }
    
    return 0;
}
